### ISSUE TITLE
Change all non-WA judgements to WA

### ISSUE THREAD

**Issue by lane55** [pc2ccs/pc2v9#635]
**Is your feature request related to a problem?** 

In the ANAC 2022 December contest, there were a number of 
RTE judgements and the shadow expected WA judgements.

This issue is intended to be a place holder for finding a simple elegant
solution to this problem.

While there is a need someday to re-design how validator
judgements are processed, until then there is a need
to create a way to ensure that certain judgemens are WA
so Shadow can match judgements.

**Feature Description**: 

Provide a way map judgement acronyms to different acronyms.

**Have you considered other ways to accomplish the same thing?** 

Currently a manual process is required.

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

Add code that would map all non-WA no's to WA by default.

**Additional context**:

**Comment by lane55** [comment]
For each place where index 2 is used as a default/initial judgment, thus
   getContest().getJudgements()[2]
Add code to lookup WA judgement, if that judgement exists use that judgement
There are 7 places in the code that use index 2 - must add code at each location.
Note that Issue #373 will refactor so there is only one location in the code that will process and assign a judgement.

**Comment by clevengr** [comment]
Strictly speaking, the judgement in this situation should be "JE" -- Judging Error (meaning, the Judging system made an error) -- In this case the error being "PC2 isn't capable of properly handling this situation".

**Comment by clevengr** [comment]
I must again record my observation/feeling that "Continuous Improvement" is an improper label for an issue like this.  This issue is basically "fix something that's not working the way it should".   By that criteria, EVERY change we do is "continuous improvement" -- so the label becomes at best superfluous and at worst confusing.  How is this "continuous improvement" and other bug fixes are not?  If I update the documentation to cover new/added features, isn't that "continuous improvement"?  For that matter, if someone files an issue proposing a NEW feature, isn't that also "continuous improvement" of the system?   I just don't get the meaning, or the purpose, for attaching this label to something like this.

**Comment by lane55** [comment]
> Strictly speaking, the judgement in this situation should be "JE" -- Judging Error (meaning, the Judging system made an error) -- In this case the error being "PC2 isn't capable of properly handling this situation".
Agreed, but out of scope (would require adding JE to judgement lists, or ensuring JE in judgement list)

**Comment by lane55** [comment]
Note that pc2 code processing CLICSValidator exit code 43, assigned result (the validator output result)
to     public static final String CLICS_WRONG_ANSWER_MSG = "Wrong Answer";
That is one reason why if "Wrong Answer" is in the judgement list, that judgement rather than index  = 2 in judgement list, is used

**Comment by lane55** [comment]
Current code, an initial judgement is assigned to index 2, getContest().getJudgements()[2], then 
code will search to match judgement display text, ex. 'result' "Wrong Answer" in the judgment list.
If no match found in judgement list then index 2 is used.
This issue will change that so if a WA exists that will be used in place of index 2.   New steps
1.  assign judgement to index 2.
2. If WA is in judgement list will assign judgement to that WA judgement
3. loop through the judgement list and if a match for the validator result string matches a judgement, assign that judgement.

**Comment by lane55** [comment]
Update reject.ini add acronyms thus, so there is a default judgement list with a WA.

Compilation Error|CE
Run-time Error|RTE
Time Limit Exceeded|TLE
Wrong Answer|WA
Excessive Output|EO
Output Format Error|OFF
Reread Problem Statement|RPS
Other - Contact Staff|CS

Otherwise the system will assign acronym (that is not WA).
See Judgement Report to see the acronym assigned, if no acronym in reject.ini

**Comment by ashoos72** [comment]
The system should not allow a user to select CLICS validation IF they have
a custom reject.ini that does not include WA.

This check should happen when defining a problem either through the GUI or
yaml load.

--Sam

**Comment by lane55** [comment]
no longer working this low priority issue
